### ISSUE TITLE
Enhancement request - TextBox to support Button and Clear Button

### ISSUE THREAD

**Issue by adragoner** [MahApps/MahApps.Metro#3164]
Currently, TextBox supports one Clear Button that can be customized. It would be nice to have a dedicated customizable Button template with an option to add Clear button next to it.

Use case:
Read only folder picker based on TextBox with button to select a folder and Clear button to clear text.

**Comment by Jeuf155** [comment]
Yes. Yes. A thousand times YES. I would love this so much. Or anyone that has programmed this manually :+1:

**Comment by Evangelink** [comment]
@punker76 As far as I see we could go either for:

1. Simplest solution: we duplicate some properties adding a number suffix. For example, we know we don't want to allow more than 2 or 3 buttons so having something like `ButtonCommand1` and `ButtonCommand2` is not so bad.

2. We don't want to have limitations and we can think of a more generic solution with a collection of a new object containing the properties linked to the buttons.

Any preference?

**Comment by timunie** [comment]
Hi @Evangelink and @punker76 ,

> *This is just my opinion / my thoughts. I am also happy with the current implementation* 

I think the second option would be my favorite, as it is more generic and can be added more easiely to different `ControlTemplates`. But it will be breaking change with a bit more `XAML` needed at the users side, or am I wrong?
I think this would also solve #3669 and #3656

Happy coding
Tim
